### PR TITLE
Picket fence calibration interpretation fix (AO-format)

### PR DESCRIPTION
**Pull request by bwmeyers**

### CODE REVIEW

**Comment by bwmeyers** [comment]
@chiamintan I know you mentioned you had some additional insight into this problem, so please feel free to comment here on what changes can/should be made :)

**Comment by chiamintan** [comment]
so the combination of line 429 : `int coarse_chan_idx = vm->coarse_chan_idxs_to_process[0];` and line 551 : `Ch = ch + coarse_chan_idx*nchan;` is what causing the bug.

What happened here is that the script looks at the observation metafits file and decide which set of coarse channel to read from the calibration solution. For cases where we input a solution with same number of channels as the vcs data for beam forming. we end up moving the pointer to read the data from away from the first coarse channel of the calibration file which corresponds to the first coarse channel of the data to be processed.

The current solution also does not work if you want to process a subset of coarse channel using a calibration solution with all channels from the observations, because the script checks the number of fine channels in the calibration solution and up/downsample it to match the number of fine channels from the subset of vcs data to process. So effectively you are using 24 coarse channels from calibration solution to process say 6 channels from vcs data, upsampled to the wrong size

**Comment by bwmeyers** [comment]
Merging this on top of the current main branch should resolve both the GPU shared memory issue and the picket fence selection issue, for the moment.

**Comment by bwmeyers** [comment]
Oh, no, actually I'll need to adjust some things...

**Comment by bwmeyers** [comment]
Okay, should square up nicely with the main branch now.  @chiamintan please take a look.

File: src/form_beam.cu [hunk 1]
```diff
@@ -255,19 +256,25 @@ __global__ void vmBeamform_kernel( cuDoubleComplex *Jv_Q,
...
+         
+            https://stackoverflow.com/questions/70765553/cuda-shared-memory-alignement-in-documentation
+
+       (Previously, the indexes were: 1*nant, 3*nant, etc.)
+     */
+    cuDoubleComplex *ex  = (cuDoubleComplex *)(&arrays[0*nant]);
+    cuDoubleComplex *ey  = (cuDoubleComplex *)(&arrays[2*nant]);
+    cuDoubleComplex *Nxx = (cuDoubleComplex *)(&arrays[4*nant]);
+    cuDoubleComplex *Nxy = (cuDoubleComplex *)(&arrays[6*nant]);
+    cuDoubleComplex *Nyy = (cuDoubleComplex *)(&arrays[8*nant]);
```
Comment (chiamintan) [inline src/form_beam.cu:277]: just want to be sure, this is the only actual change right? Others seems to just be debugging messages and comments

Comment (bwmeyers) [reply src/form_beam.cu:277]: Yes, that's correct.

**Review by chiamintan — APPROVED** [review]
Aight let's go

**Comment by bwmeyers** [comment]
Just going to see if I can make those compile/Codacy warnings go away.

**Pull request merged**